packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts

```diff
@@ -473,17 +474,27 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumerBase {
             }
         }
 
+        // Return any unspent sell tokens.
+        const payTakerTokens = [sellToken];
+        // Return any unspent intermediate tokens for two-hop swaps.
+        if (quote.isTwoHop) {
+            payTakerTokens.push(intermediateToken);
+        }
+        // Return any unspent ETH. If ETH is the buy token, it will
+        // be returned in TransformERC20Feature rather than PayTakerTransformer.
+        if (!isToETH) {
+            payTakerTokens.push(ETH_TOKEN_ADDRESS);
+        }
         // The final transformer will send all funds to the taker.
         transforms.push({
             deploymentNonce: this.transformerNonces.payTakerTransformer,
             data: encodePayTakerTransformerData({
-                tokens: [sellToken, buyToken, ETH_TOKEN_ADDRESS].concat(quote.isTwoHop ? intermediateToken : []),
+                tokens: payTakerTokens,
                 amounts: [],
             }),
         });
-
-        const calldataHexString = this._exchangeProxy
-            .transformERC20(
+        const calldataHexString = new TransformERC20FeatureContract(this._exchangeProxy.address, FAKE_PROVIDER)
```

Comment:
We should pull this contract instance out into a const. Contract wrappers are very heavy objects.